frame/multisig/src/lib.rs

```diff
@@ -260,103 +363,32 @@ decl_module! {
 		///   `DepositBase + threshold * DepositFactor`.
 		/// -------------------------------
 		/// - Base Weight:
-		///     - Create: 46.55 + 0.089 * S µs
-		///     - Approve: 34.03 + .112 * S µs
-		///     - Complete: 40.36 + .225 * S µs
+		///     - Create:          41.89 + 0.118 * S + .002 * Z µs
+		///     - Create w/ Store: 53.57 + 0.119 * S + .003 * Z µs
+		///     - Approve:         31.39 + 0.136 * S + .002 * Z µs
+		///     - Complete:        39.94 + 0.26  * S + .002 * Z µs
 		/// - DB Weight:
-		///     - Reads: Multisig Storage, [Caller Account]
-		///     - Writes: Multisig Storage, [Caller Account]
+		///     - Reads: Multisig Storage, [Caller Account], Calls (if `store_call`)
+		///     - Writes: Multisig Storage, [Caller Account], Calls (if `store_call`)
 		/// - Plus Call Weight
 		/// # </weight>
-		#[weight = (
-			weight_of::as_multi::<T>(other_signatories.len(), call.get_dispatch_info().weight),
-			call.get_dispatch_info().class,
-			Pays::Yes,
+		#[weight = weight_of::as_multi::<T>(
+			other_signatories.len(),
+			call.len(),
+			*max_weight,
+			true,
+			true,
```

Comment:
Was the dispatch class changed intentionally?

Reply:
also a nit: I don't particularly like the two bool params without context here (need to check the function above to figure out the meaning and whether the ordering is correct)

Reply:
We can't know the dispatch class at time of dispatch since the call is a vector. It would require us to encode the call.